### DISCUSSION TITLE
v4.0.0

### DISCUSSION THREAD

**Discussion by yanue** [yanue/V2rayU#1326] [General]
改进ping,用真实代理计算响应时间
 端口被占用问题,弹窗提醒修改
 按订阅分组显示服务器列表
 改进pac生成(支持手动编辑user-rules.txt)
 改进非标准url链接无法导入问题
 集成appcenter.ms(崩溃日志收集)
 去掉v2ray-core自动更新功能
 新增grpc协议
 reality最新
 支持clash订阅(不支持clash规则)
 代理情况下更新代理和pac
 系统代理不做backup和restore

<hr /><em>This discussion was created from the release <a href='https://github.com/yanue/V2rayU/releases/tag/v4.0.0'>v4.0.0</a>.</em>

**Comment by 0f7b90** [comment]
mac m1 sonoma，安装运行之后不显示任何界面，也没有状态栏图标，任务管理器里有v2rayu的进程，但没有v2ray核心进程

**Reply by yanue** [reply]
重启也一样吗

**Reply by 0f7b90** [reply]
解决了，先安装旧版本再安装新版就好了

**Reply by kulithDev** [reply]
@CometYZ @yanue  Is it working on Sanoma? I tried adding a vmess and tried tunning but it didn't work

**Comment by icoolding** [comment]
@yanue 更新完一直弹 4.0.3的更新。安装更新后也不行。
![截屏2023-10-15 11 15 16](https://github.com/yanue/V2rayU/assets/57248537/249b4464-8b29-486c-a9df-1aeef0e5bf95)
![截屏2023-10-15 11 13 49](https://github.com/yanue/V2rayU/assets/57248537/1ba80d2a-211f-4b60-93a4-001f3faa8bba)

**Reply by yanue** [reply]
暂时跳过

**Reply by icoolding** [reply]
ok，坐等正式版

**Reply by icoolding** [reply]
@yanue 今天开始闪退了。

**Comment by wallinwind** [comment]
3.8上有些正常的vless节点，升级4.0后不可用，用小火箭测了节点是好的，退回后又可用了。套TF的vless

请修复。

**Comment by wolgy** [comment]
ping speed vmess节点正常，vless节点只显示-1。多次重试都是这样

**Comment by wolgy** [comment]
routing rule中设置了block，但是被设置block的domain还是能够direct访问。这个issue似乎没有得到解决 https://github.com/yanue/V2rayU/issues/1301

**Comment by OpenNEONE** [comment]
4.0 ，千呼万唤始出来。 期待v2rayU越走越远！👏

**Comment by bughda1995** [comment]
macOS Monterey 启动core时报错  “端口已被占用，http端口1087已被使用，请更换。” 怎么解决？

**Comment by onshine** [comment]
装完4.0，不能科学上网，又换回3.8

**Comment by kulithDev** [comment]
@yanue any versions are not working on new mac sanoma os. Anyway facing the same issue?

**Reply by zhangxiaoyu2046** [reply]
v3.8.0 can work on my Sonoma 14.2, maybe try re-installing the stable version?

**Comment by cchuckliao** [comment]
用不了，打开没有任何反应。用命令行打开提示缺少文件
Failed to start: main: failed to load config files: [./config.json] > infra/conf/serial: failed to read config: ./config.json > open ./config.json: no such file or directory
然后换到3.8也是这样。没道理，我就是正常的安装过程呢

**Reply by jixing475** [reply]
同问, 同样的问题(macos Ventura 13.4)

**Reply by jixing475** [reply]
真是要命, 更新后出了问题, 结果装回 3.8 出现同样的问题, 现在不知道咋办了:
start v2ray fail : v2ray config not found

**Reply by icoolding** [reply]
卸载干净再重装。

**Comment by caily111** [comment]
感谢更新，4.0终于可以关闭代理后，网络设置里的代理可以自动关闭了。

**Comment by MrCandyfloss** [comment]
Would you plz update the App icon in the 4.0 version, which will go with the latest trend of macOS instead of the current round one?
